baseplate/__init__.py

```diff
@@ -465,35 +491,35 @@ def start(self):
         for observer in self.observers:
             observer.on_start()
 
-    def set_tag(self, key, value):
+    def set_tag(self, key: str, value: Any) -> None:
         """Set a tag on the span.
 
         Tags are arbitrary key/value pairs that add context and meaning to the
         span, such as a hostname or query string. Observers may interpret or
         ignore tags as they desire.
 
-        :param str key: The name of the tag.
-        :param value: The value of the tag, must be a string/boolean/number.
```

Comment:
Is this restriction no longer true? If it is, I think you could use a Union to enforce it if it is rather than just using `Any`.

Reply:
The current tracing implementation accepts bools, ints, and anything with `__str__`. We're placing some exception instances as tags right now. Could clean that up and tighten the param type, or leave as-is since at worst we'll just get a stringy version of whatever's passed in (I should definitely update the docs if we go the latter, good catch).